### ISSUE TITLE
Cross node routing not working

### ISSUE THREAD

**Issue by maaeps** [projectcalico/calico#2417]
As I want to add IPv6 support, I switched from flannel to calico.
I am finally at a point, where my pods at least start. Unfortunately cross node routing is not working. I a not an expert in network administration but I have done my very best to help myself.

In my setup, there is one masternode with the pod ip range of 10.244.0.0/24 and a worker node with 10.244.1.0/24.
# Expected Behavior

Normally I should be able to connect/ping to a pod running on an other node.

## Current Behavior

Currently, pinging a pod running on an other node gives me no output and looks like it takes forever. I think I come in a loop while routing.

## Possible Solution

I really hope you can help me.

## Steps to Reproduce (for bugs)

I don't really know how this happened, but I setup my cluster at first with kubeadm (flannel) and used then [this](https://docs.projectcalico.org/v3.5/getting-started/kubernetes/installation/calico) guid to move to callico.
I hope, I just forgot to setup one part.

## Context

I began to realize the issue, as I us a pv provisioner and can't connect cross node to the NFS Server running in a pod.

## Your Environment

* Calico version: 3.5.1
* Orchestrator version (e.g. kubernetes, mesos, rkt): kubernetes
* Operating System and version: fedora 29 (master), ubuntu 17.10

As I do not know what files might be interesting for you, I hope you will tell me.

Thanks for your help,
Maximilian

**Comment by robododge** [comment]
I am having the same issue with Calico 3.4.1 in a locally deployed Kubernetes, Rhel 7.5.

Can you please provide the listing of your routing table and the current list of interfaces. And Iptables chains?

Output from the following:

1. ip route
2. ip link
3. iptables-save

**Comment by maaeps** [comment]
Well, I found the solution...
"projectcalico.org/IPv4Address" was not set correctly on one node.

**Comment by robododge** [comment]
My issue resolved as well.   I was installing Kubernetes using Kubespary.  Kubespray was setting up the Calico cni-plugin config (in /etc/cni/net.d) with the FQDN of the node.  Then dual entries for ipv4/block ended up in etcd, one block for the key with FQDN, and another block with the key for the node hostname. 
```
/calico/ipam/v2/host/zlt10878.example.com/ipv4/block/10.246.254.192-26
/calico/ipam/v2/host/zlt10878/ipv4/block/10.244.103.0-26
```
routes for all the nodes in my cluster were setup using the 10.244.103.0/26 subnet, but IP addresses were being assigned. to pods using addresses from the 0.246.254.192/26 subnet.

I fixed it by stopping Calico-node pods by removing the daemonset and saving in daemonset.yml file.  Then, on each node, I changed the incorrect fqdn nodename setting to the non qualified hostname  in the /etc/cni/net.d/calico.conflist.template and /etc/cni/net.d/10-calico.conflist files.   Then, deleted all the routes and interfaces created by calico (ip route del /  ip link del).  Finally, i started Calico pods back by re-applying the daemonset.yml.   At that point, routes and IP assignments created by Calico started matching correctly.
